### ISSUE TITLE
[Feat]: Number of cores / CPUs as metric

### ISSUE THREAD

**Issue by jaroslawr** [netdata/netdata#15150]
### Description

It looks like netdata does not expose the number of CPU cores / CPUs as a metric, only as a label. How much of a problem this is depends what you then do with the metrics, with Graphite and Grafana this means I am not able to draw the number of CPUs as a line on a system load plot, for example. On the other hand, the amount of RAM in the system is exposed as a metric.

### Importance

really want

### Value proposition

For such basic information it makes sense to expose it both as a label and as a metric.

**Comment by jaroslawr** [comment]
(Actually I think I was wrong, the total amount of memory also is NOT available as a metric, would be nice to have IMO)

**Comment by k0ste** [comment]
Actually `netdata` exports this information, you can create record_rule by yourself, for example if you need gigabytes:

```
round(sum by (instance) (netdata_system_ram_MiB_average / 1000))
```
<img width="1245" alt="Screenshot 2023-06-06 at 21 17 13" src="https://github.com/netdata/netdata/assets/7759548/7c9e58c6-7cf6-4943-adad-bdfecffd9102">

**Comment by jaroslawr** [comment]
@k0ste I am using the graphite exporter, you are right it is possible to aggregate the existing metrics to get the CPU count, but not very convenient, I think it would be nice to simply have a dedicated metric with the number of CPUs and the amount of RAM (for example pcp which we use as another metrics collector exposes this).

**Comment by ilyam8** [comment]
Hey, @jaroslawr. Will `info` metric do?

If you

```
# exporting.conf
[prometheus:exporter]
    send automatic labels = yes
```

you get all host labels in the `netdata_info` metrics including the number of cores, e.g.

```
netdata_info{instance="pve-deb-work",application="netdata",version="v1.40.0-48-g705ebd86c",_cloud_provider_type="unknown",_cloud_instance_type="unknown",_cloud_instance_region="unknown",_os_name="Debian GNU/Linux",_os_version="12 (bookworm)",_kernel_version="6.1.0-9-amd64",_system_cores="16",_system_cpu_freq="1999000000",_system_ram_total="33703321600",_system_disk_space="214748364800",_architecture="x86_64",_virtualization="kvm",_container="none",_container_detection="systemd-detect-virt",_virt_detection="systemd-detect-virt",_is_k8s_node="false",_install_type="custom",_aclk_available="true",_mqtt_version="5",_aclk_proxy="none",_aclk_ng_new_cloud_protocol="true",_is_ephemeral="false",_has_unstable_connection="false",_is_parent="false"} 1 1687957156276
```

**Comment by jaroslawr** [comment]
@ilyam8 I am using the graphite exporter - is there anything equivalent in this case?

**Comment by ilyam8** [comment]
Not really, it is possible to `send automatic labels = yes` for graphite but it will add all the labels as tags to **every** exported metric.

**Comment by ilyam8** [comment]
@thiagoftsm can we fix this? I think we need:

- add `netdata_info` metric to graphite.
- don't add `host_labels` to other metrics.

The same we do in prometheus exporter.

**Comment by ilyam8** [comment]
> Not really, it is possible to send automatic labels = yes for graphite but it will add all the labels as tags to every exported metric.

Example:


```
netdata.pve-deb-work.net.ens18.received;_cloud_provider_type=unknown;_cloud_instance_type=unknown;_cloud_instance_region=unknown;_os_name=Debian_GNU/Linux;_os_version=12_(bookworm);_kernel_version=6.1.0-9-amd64;_system_cores=16;_system_cpu_freq=1999000000;_system_ram_total=33703321600;_system_disk_space=214748364800;_architecture=x86_64;_virtualization=kvm;_container=none;_container_detection=systemd-detect-virt;_virt_detection=systemd-detect-virt;_is_k8s_node=false;_install_type=custom;_aclk_available=true;_mqtt_version=5;_aclk_proxy=none;_aclk_ng_new_cloud_protocol=true;_is_ephemeral=false;_has_unstable_connection=false;_is_parent=false 0.0000000 1689076026
```

The list of labels:

- _cloud_provider_type=unknown
- _cloud_instance_type=unknown
- _cloud_instance_region=unknown
- _os_name=Debian_GNU/Linux
- _os_version=12_(bookworm)
- _kernel_version=6.1.0-9-amd64
- _system_cores=16
- _system_cpu_freq=1999000000
- _system_ram_total=33703321600
- _system_disk_space=214748364800
- _architecture=x86_64
- _virtualization=kvm
- _container=none
- _container_detection=systemd-detect-virt
- _virt_detection=systemd-detect-virt
- _is_k8s_node=false
- _install_type=custom
- _aclk_available=true
- _mqtt_version=5
- _aclk_proxy=none
- _aclk_ng_new_cloud_protocol=true
- _is_ephemeral=false
- _has_unstable_connection=false
- _is_parent=false

> We have "configured" host labels (set in netdata.conf in [host labels]) and automatic (added on start up after host info detection)

@jaroslawr since you are a graphite user (@k0ste please share your opinion too): 

- I guess this is wrong and doesn't make sense to add automatic host labels to every metric and this is a bug. Am I right?
- Will adding the equivalent of Prometheus "netdata_info" metrics (that will contain all automatic host labels (the list above)) to graphite exporting do?

**Comment by k0ste** [comment]
> I guess this is wrong and doesn't make sense to add automatic host labels to every metric and this is a bug. Am I right?

Definitely, one meta label is enough, and can be joined eventually (see https://github.com/prometheus-community/smartctl_exporter/pull/83)

But, it will be nice, if this labels exports by default, without `send automatic labels` option

**Comment by thiagoftsm** [comment]
Hello @k0ste ,

I liked your suggestion and I set send labels enabled by default for `graphite` with my last commit for PR #15334. 
I plan to evaluate the situation with other users for other exporters.

Best regards!

**Comment by jaroslawr** [comment]
@ilyam8 It's not something I am personally interested in, but it's not completely pointless either to add these labels as tags, you can use these to filter series, e.g. plot some series like CPU load only for hosts that have number_of_cores == 16. However to put the number of cores on the graph in Grafana as a line (my use case) you need to have a metric rather than a tag. So I can imagine having this information in both forms: as tags and/or as metrics to be useful.

**Comment by thiagoftsm** [comment]
Hello @k0ste ,

We use underscore when labels are detected by `netdata` only. We did not want to add a word as prefix to avoid big changes in the label meaning.

Best regards!

**Comment by k0ste** [comment]
@thiagoftsm good, thanks!

Also, I have another request: we need processor model, to determine what exactly CPU installed at the host. Seems `NETDATA_SYSTEM_CPU_MODEL` is already present - can we add it to `sqlite_metadata`?

**Comment by thiagoftsm** [comment]
> @thiagoftsm good, thanks!
> 
> Also, I have another request: we need processor model, to determine what exactly CPU installed at the host. Seems `NETDATA_SYSTEM_CPU_MODEL` is already present - can we add it to `sqlite_metadata`?

Hello @k0ste ,

I agree 100% with you, but I will not use the same PR,  I will talk tomorrow with @stelfrag and @MrZammler about your suggestion for we bring ti ASAP. :handshake: .

Best regards!

**Comment by k0ste** [comment]
> > @thiagoftsm good, thanks!
> > Also, I have another request: we need processor model, to determine what exactly CPU installed at the host. Seems `NETDATA_SYSTEM_CPU_MODEL` is already present - can we add it to `sqlite_metadata`?
> 
> Hello @k0ste ,
> 
> I agree 100% with you, but I will not use the same PR, I will talk tomorrow with @stelfrag and @MrZammler about your suggestion for we bring it ASAP. 🤝 .
> 
> Best regards!

Did you manage to agree on something?

**Comment by thiagoftsm** [comment]
Hello @k0ste ,

This project hasn’t been forgotten; we’ve had to work other features in the past few months. We’ll post updates here as soon as we have them.

Best regards!

**Comment by k0ste** [comment]
> Do you mean we need to add this to the `netdata_info` labels?

Yes. Or, to another metric, like `netdata_cpu_info`, to avoid problems with labels limit in some cases

**Comment by k0ste** [comment]
Netdata 2.1.1 provided

----

netdata_info{instance="infra1.opentech.local",application="netdata",version="v2.1.1",_cloud_provider_type="unknown",_cloud_instance_type="unknown",_cloud_instance_region="unknown",_os_name="CentOS Stream",_os_version="8",_kernel_version="4.18.0-552.3.1.el8.x86_64",**_system_cores**="4",_system_cpu_freq="3000000000",**_system_cpu_model**="Intel(R) Xeon(R) CPU E3-1220 v5 @ 3.00GHz",_system_ram_total="67081297920",_system_disk_space="12962558214144",_architecture="x86_64",_virtualization="none",_container="none",_container_detection="systemd-detect-virt",_virt_detection="systemd-detect-virt",_is_k8s_node="false",_install_type="custom",_aclk_available="true",_mqtt_version="5",_aclk_proxy="none",_aclk_ng_new_cloud_protocol="true",_is_ephemeral="false",_has_unstable_connection="false",_is_parent="false",_hostname="example.com",_os="linux"}

----

Seems issue was resolved

**Comment by mabed-fr** [comment]
I'm on netdata 2.2.6 and i have the metrics:

netdata_active_processors

;)
